flow360/accounts_utils.py

```diff
@@ -0,0 +1,166 @@
+                )
+                if value == "q":
+                    return None
+                if int(value) in range(0, user_count):
+                    return int(value)
+                print(f"Value out of range [0 - {user_count - 1}]")
+                continue
+            except ValueError:
+                print("Invalid input type, please input an integer value:")
+                continue
+
+    # Requires fixing from the backend side - support for portal webapi calls with apikey authentication
+    @staticmethod
+    def _get_supported_users():
+        try:
+            response = http.portal_api_get("auth")
+            access = response.json()["data"]
+            keys = access["user"]
+            supported_users = keys["guestUsers"]
+            if supported_users:
+                return supported_users
+            return []
+        except HTTPError as error:
+            raise WebError("Failed to retrieve supported user data from server") from error
+
+    @staticmethod
+    def _get_company_users():
+        try:
+            response = http.get("flow360/account")
+            company_users = response["tenantMembers"]
+            if company_users:
+                return company_users
+            return []
+        except HTTPError as error:
+            raise WebError("Failed to retrieve company user data from server") from error
+
+    def _check_state_consistency(self):
+        if Env.impersonate != self._current_user_identity:
+            log.warning(
+                (
+                    f"Environment impersonation ({Env.impersonate}) does "
+                    f"not match current account ({self._current_user_identity}), "
+                    "this may be caused by explicit modification of impersonation "
+                    "in the environment, use choose_shared_account() instead."
+                )
+            )
+            self._current_email = None
+            self._current_user_identity = None
+
+    def is_confirmed(self):
+        """check if the user confirmed that he wants to submit resources to a shared account"""
+        return self._confirmed_submit
+
+    def confirm_submit(self):
+        """confirm submit for the current session"""
+        self._confirmed_submit = True
+
+    def choose_shared_account(self, email=None):
+        """choose a shared account to impersonate
+
+        Parameters
+        ----------
+        email : str, optional
+           user email to impersonate (if email exists among shared accounts),
+           if email is not provided user can select the account interactively
+        """
+        company_users = self._get_company_users()
+
+        if len(company_users) == 0:
+            log.info("There are no accounts shared with the current user")
+            return
+
+        selected = None
+
+        addresses = [d["userEmail"] for d in company_users]
```

Comment:
we try to avoid single letter variables, even in such instances. Use `user` instead

Reply:
Renamed to user in 644f8ee